src/apps/genie/transcode-sam/utils.cc

```diff
@@ -21,6 +21,9 @@ bool compare(const genie::core::record::Record& r1, const genie::core::record::R
     if (r2.getAlignments().empty()) {
         return true;
     }
+    if (r1.getAlignmentSharedData().getSeqID() != r2.getAlignmentSharedData().getSeqID()) {
```

Comment:
Why do we have this additional condition for sorting?

Reply:
One of the optimizations in this pull request is to sort by sequence and position at the same time. We have no separate files for chromosomes anymore - instead the chromosomes are just considered to be part of the position.

1. Chromosomes differ, use chromosomes to infer the order of records
2. If chromosomes are different, use position on the chromosome instead